nemo/collections/asr/data/ssl_dataset.py

```diff
@@ -0,0 +1,723 @@
+
+
+def _parse_manifest_item(line: str, manifest_file: str) -> Dict[str, Any]:
+    """
+    Specialized function to parse the manifest file by ignoring text,
+    such that nemo dataset can save time on tokenizing text.
+    """
+    item = json.loads(line)
+
+    # Audio file
+    if 'audio_filename' in item:
+        item['audio_file'] = item.pop('audio_filename')
+    elif 'audio_filepath' in item:
+        item['audio_file'] = item.pop('audio_filepath')
+    else:
+        raise KeyError(f"No 'audio_filename' or 'audio_filepath' in manifest item: {item}")
+
+    item['audio_file'] = get_full_path(audio_file=item['audio_file'], manifest_file=manifest_file)
+
+    # Duration.
+    if 'duration' not in item:
+        item['duration'] = None
+
+    # dummy text
+    item['text'] = ""
+
+    item = dict(
+        audio_file=item['audio_file'],
+        duration=item['duration'],
+        text=item['text'],
+        offset=item.get('offset', None),
+        speaker=item.get('speaker', None),
+        orig_sr=item.get('orig_sample_rate', None),
+        token_labels=item.get('token_labels', None),
+        lang=item.get('lang', None),
+    )
+    return item
+
+
+def _audio_noise_collate_fn(batch: List[AudioNoiseItem], batch_augmentor: Any = None) -> AudioNoiseBatch:
+    audios = [x.audio for x in batch]
+    audio_lengths = [x.audio_len for x in batch]
+    max_audio_len = max(audio_lengths).item()
+
+    noises = [x.noise for x in batch]
+    noise_lengths = [x.noise_len for x in batch]
+
+    noisy_audios = [x.noisy_audio for x in batch]
+    noisy_audio_lengths = [x.noisy_audio_len for x in batch]
+
+    audio_signal_list = []
+    noise_signal_list = []
+    noisy_audio_signal_list = []
+    for i, audio in enumerate(audios):
+        audio_len = audio.size(0)
+        if audio_len < max_audio_len:
+            pad = (0, max_audio_len - audio_len)
+            audio = torch.nn.functional.pad(audio, pad)
+        audio_signal_list.append(audio)
+
+        noise = noises[i]
+        noise_len = noise.size(0)
+        if noise_len < max_audio_len:
+            pad = (0, max_audio_len - noise_len)
+            noise = torch.nn.functional.pad(noise, pad)
+        noise_signal_list.append(noise[:max_audio_len])
+
+        noisy_audio = noisy_audios[i]
+        noisy_audio_len = noisy_audio.size(0)
+        if noisy_audio_len < max_audio_len:
+            pad = (0, max_audio_len - noisy_audio_len)
+            noisy_audio = torch.nn.functional.pad(noisy_audio, pad)
+        noisy_audio_signal_list.append(noisy_audio[:max_audio_len])
+
+    audio_signal = torch.stack(audio_signal_list).float()
+    audio_lengths = torch.stack(audio_lengths).long()
+    noise_signal = torch.stack(noise_signal_list).float()
+    noise_lengths = torch.stack(noise_lengths).long()
+    noisy_audio_signal = torch.stack(noisy_audio_signal_list).float()
+    noisy_audio_lengths = torch.stack(noisy_audio_lengths).long()
+
+    output = AudioNoiseBatch(
```

Comment:
Do note that DataClass objects are NOT automatically mapped to correct device by PTL

Reply:
yes I use the `move_data_to_device()` function to handle that